convert/convert_granite.go

```diff
@@ -0,0 +1,101 @@
+package convert
+
+import (
+	"cmp"
+	"math"
+
+	"github.com/ollama/ollama/llm"
+)
+
+type graniteModel struct {
+	llamaModel
+
+	ResidualScale  float32 `json:"residual_multiplier"`
+	EmbeddingScale float32 `json:"embedding_multiplier"`
+	AttentionScale float32 `json:"attention_multiplier"`
+	LogitsScale    float32 `json:"logits_scaling"`
+}
+
+var _ ModelConverter = (*graniteModel)(nil)
+
+func (p *graniteModel) KV(t *Tokenizer) llm.KV {
+	kv := p.ModelParameters.KV(t)
+	kv["general.architecture"] = "granite"
+	kv["granite.vocab_size"] = p.VocabSize
+
+	kv["granite.block_count"] = cmp.Or(p.NLayers, p.NumHiddenLayers, p.NLayer)
+
+	if contextLength := cmp.Or(p.MaxPositionEmbeddings, p.NCtx); contextLength > 0 {
```

Comment:
I don't think you need each of the branching statements here. Just:

```
kv["granite.context_length"] = cmp.Or(p.MaxPositionEmbeddings, p.NCtx)
```

should be sufficient.